### PR TITLE
Digital twin/telemetry explosion

### PR DESCRIPTION
**Pull request by davilu**
# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-java/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: digital-twin/device/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponent.java [hunk 1]
```diff
@@ -59,17 +59,27 @@ protected AbstractDigitalTwinComponent(@NonNull String digitalTwinComponentName,
      * The application may invoke this function as many times as it wants, subject to the device's resource availability. The application does NOT need to wait for each send to finish.
      * The SDK will automatically attempt to retry the telemetry send on transient errors.
      *
+     * @param payload DigitalTwin telemetry message payload. It should be a valid json string represent a map. The key is property name. the JSON should look like:
+     *				{
```
Comment (drwill-ms) [inline digital-twin/device/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponent.java]: tabbing looks off on this brace

File: digital-twin/device/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponent.java [hunk 2]
```diff
@@ -80,10 +90,21 @@ protected AbstractDigitalTwinComponent(@NonNull String digitalTwinComponentName,
      * The device application calls this function to send telemetry to the server. The call will be blocked and will return the result once telemetry is processed.
      * The SDK will automatically attempt to retry the telemetry send on transient errors.
      *
+     * @param payload DigitalTwin telemetry message payload. It should be a valid json string represent a map. The key is property name. the JSON should look like:
+     *				{
```
Comment (drwill-ms) [inline digital-twin/device/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponent.java]: here too

File: digital-twin/device/src/test/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponentTest.java [hunk 3]
```diff
@@ -41,8 +41,7 @@
 public class AbstractDigitalTwinComponentTest {
     private static final String DIGITAL_TWIN_COMPONENT_NAME = "DIGITAL_TWIN_COMPONENT_NAME";
     private static final String DIGITAL_TWIN_INTERFACE_ID = "DIGITAL_TWIN_INTERFACE_ID";
-    private static final String TELEMETRY_NAME = "TELEMETRY_NAME";
-    private static final String TELEMETRY_PAYLOAD = "TELEMETRY_PAYLOAD";
+    private static final String TELEMETRY_PAYLOAD = "{\"TELEMETRY_NAME\": \"TELEMETRY_PAYLOAD\"}";
```
Comment (drwill-ms) [inline digital-twin/device/src/test/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponentTest.java]: The value of the telemetry is the same name as the entire thing, which is confusing. Suggest changing this to TELEMETRY_VALUE.

File: digital-twin/device/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponent.java [hunk 4]
```diff
@@ -59,17 +59,27 @@ protected AbstractDigitalTwinComponent(@NonNull String digitalTwinComponentName,
      * The application may invoke this function as many times as it wants, subject to the device's resource availability. The application does NOT need to wait for each send to finish.
      * The SDK will automatically attempt to retry the telemetry send on transient errors.
      *
+     * @param payload DigitalTwin telemetry message payload. It should be a valid json string represent a map. The key is property name. the JSON should look like:
```
Comment (drwill-ms) [inline digital-twin/device/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponent.java]: grammar: representing

Comment (drwill-ms) [inline digital-twin/device/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponent.java]: grammar: capital T in "the" from "the JSON should look like:"

File: digital-twin/device/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponent.java (hunk 1 above)
Comment (davilu) [reply digital-twin/device/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponent.java]: fixed

File: digital-twin/device/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponent.java (hunk 2 above)
Comment (davilu) [reply digital-twin/device/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponent.java]: fixed

File: digital-twin/device/src/test/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponentTest.java (hunk 3 above)
Comment (davilu) [reply digital-twin/device/src/test/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponentTest.java]: fixed

File: digital-twin/device/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponent.java (hunk 4 above)
Comment (davilu) [reply digital-twin/device/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponent.java]: fixed

Comment (davilu) [reply digital-twin/device/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/device/AbstractDigitalTwinComponent.java]: fixed

**Pull request merged**